starter-code/pizza.js

```diff
@@ -1,85 +1,194 @@
+  renderButtons();
+  renderPrice();
 }
 
 function renderPepperonni() {
-  document.querySelectorAll('.pep').forEach(function($pep){
+  document.querySelectorAll('.pep').forEach(function($pep) {
     if (state.pepperonni) {
-      $pep.style.visibility = "visible";
+      $pep.style.visibility = 'visible';
+    } else {
+      $pep.style.visibility = 'hidden';
     }
-    else {
-      $pep.style.visibility = "hidden";
-    }
-  })
+  });
 }
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(function($mush) {
+    if (state.mushrooms) {
+      $mush.style.visibility = 'visible';
+    } else {
+      $mush.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGreenPeppers() {
-  // Iteration 1: set the visibility of `<section class="green-pepper">`
+  // Iteration 1: set the visibility of `<section class="green-pepper">
+  document.querySelectorAll('.green-pepper').forEach(function($greenPep) {
+    if (state.greenPeppers) {
+      $greenPep.style.visibility = 'visible';
+    } else {
+      $greenPep.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  document.querySelectorAll('.sauce').forEach(function($sauce) {
+    if (state.whiteSauce) {
+      $sauce.classList.add('sauce-white');
+    } else {
+      $sauce.classList.remove('sauce-white');
+    }
+  });
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+
+  document.querySelectorAll('.crust').forEach(function($crust) {
+    if (state.glutenFreeCrust) {
+      $crust.classList.add('crust-gluten-free');
+    } else {
+      $crust.classList.remove('crust-gluten-free');
+    }
+  });
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+
+  /* var state = {
+    pepperonni: true,
+    mushrooms: true,
+    greenPeppers: true,
+    whiteSauce: false,
+    glutenFreeCrust: false,
+  }; */
+
+  //go through every state key (Object.key)
+
+  Object.keys(state).forEach((item) => {
+    // check the state
+    // console.log(item, state[item], classMatchState[item]);
+    //select button
+    let btn = document.querySelector('.' + classMatchState[item]);
+    // console.log(btn);
+
+    //if state of item is true, than
+    if (state[item]) {
+      //add active
+      btn.classList.add('active');
+    } else {
+      //remove active
+      btn.classList.remove('active');
+    }
+  });
 }
+const showOrHidePriceElement = (listElement, show) => {
+  document.querySelectorAll('.panel.price > ul > li').forEach((element) => {
+    if (
+      element.innerHTML.indexOf(ingredients[listElement].name.toLowerCase()) !==
+      -1
+    ) {
+      // console.log(element.style.visibility);
+      element.style.visibility = show ? 'visible' : 'hidden';
```

Comment:
Using the visible/hidden style will cause a white hole to appear on your webpage when removing Mushrooms for instance. This can be avoided if you rather create a string where you just add the text:
let string = ""
if(state.pepperonni){string += "<li>$1 Pepperonni</li>} else if (state.mushrooms){string += "<li>$1 Mushrooms</li>" ....

and then just set the innerHTML as this string in the end. This will not create an empty space.